javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +520,356 @@ object Character {
+    private[this] val blocksByNormalizedName = new java.util.HashMap[String, UnicodeBlock]()
+
+    private[this] def addNameAliases(properName: String, block: UnicodeBlock): Unit = {
+      // Add normalized aliases
+      val lower = properName.toLowerCase
+      //   lowercase + spaces
+      blocksByNormalizedName.put(lower, block)
+      //   lowercase + no spaces
+      blocksByNormalizedName.put(lower.replaceAll("\\s", ""), block)
+      //   tokens as _
+      blocksByNormalizedName.put(lower.replaceAll("[\\s\\-_]", "_"), block)
+    }
+
+    private[this] def addUnicodeBlock(name: String, properName: String, start: Int, end: Int): UnicodeBlock =  {
+      val block = new UnicodeBlock(name, start, end)
+      allBlocks.add(block)
+      addNameAliases(properName, block)
+      block
+    }
+
+    // Special constants
+    val SURROGATES_AREA = new UnicodeBlock("SURROGATES_AREA",0x0, 0x0)
+    blocksByNormalizedName.put("SURROGATES AREA".toLowerCase, SURROGATES_AREA)
+    blocksByNormalizedName.put("SURROGATES_AREA".toLowerCase, SURROGATES_AREA)
+
+    val BASIC_LATIN = addUnicodeBlock("BASIC_LATIN", "Basic Latin", 0x0000, 0x007F)
+    val LATIN_1_SUPPLEMENT = addUnicodeBlock("LATIN_1_SUPPLEMENT", "Latin-1 Supplement", 0x0080, 0x00FF)
+
+    // scalastyle:off line.size.limit
+/*
+    // Get the "proper name" for JVM block name
+    val blockNameMap: Map[String, String] = {
+      val blocksSourceURL = new java.net.URL("http://unicode.org/Public/UCD/latest/ucd/Blocks.txt")
+      val source = scala.io.Source.fromURL(blocksSourceURL, "UTF-8")
+      source
+        .getLines()
+        .filterNot {
+          _.startsWith("#")
+        }
+        .flatMap { line =>
+          line.split(';') match {
+            case Array(_, name) =>
+              val trimmed = name.trim
+              val jvmName = trimmed.replaceAll(raw"[\s\-_]", "_").toUpperCase
+              Some(jvmName -> trimmed)
+            case _ => None
+          }
+        }.toMap ++ Seq(
+          "GREEK" -> "Greek",
+          "CYRILLIC_SUPPLEMENTARY" -> "Cyrillic Supplementary",
+          "COMBINING_MARKS_FOR_SYMBOLS" -> "Combining Marks For Symbols"
+        )
+    }
+
+    val blocksAndCharacters = (256 to Character.MAX_CODE_POINT)
+      .map { cp => Character.UnicodeBlock.of(cp) -> cp }.filterNot{ _._1 == null }
+
+    val orderedBlocks = blocksAndCharacters.map{ _._1 }.distinct
+
+    val blockLowAndHighCodePointsMap = {
+      blocksAndCharacters.groupBy{ _._1 }.mapValues { v =>
+        val codePoints = v.map{ _._2 }
+        (codePoints.min, codePoints.max)
+      }
+    }
+
+    orderedBlocks.foreach{ b =>
+      val minCodePoint = "%04x".format(blockLowAndHighCodePointsMap(b)._1).toUpperCase
+      val maxCodePoint = "%04x".format(blockLowAndHighCodePointsMap(b)._2).toUpperCase
+
+      println(s"""  val $b = addUnicodeBlock("$b", "${blockNameMap.getOrElse(b.toString, throw new IllegalArgumentException("$b"))}", 0x$minCodePoint, 0x$maxCodePoint)""")
```

Comment:
To avoid the long line:
```suggestion
      val properName = blockNameMap.getOrElse(b.toString, throw new IllegalArgumentException("$b"))
      println(s"""  val $b = addUnicodeBlock("$b", "$properName", 0x$minCodePoint, 0x$maxCodePoint)""")
```